packages/server/src/auth/auth.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable } from '@nestjs/common';
+import * as bcrypt from 'bcrypt';
+import { PrismaService } from '../prisma/prisma.service';
+import { UsersService } from '../users/users.service';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class AuthService {
+  constructor(private usersService: UsersService, private prisma: PrismaService, private jwtService: JwtService) {}
+
+  async validateUsername(username: string, password: string): Promise<any> {
+    const user = await this.usersService.findOneUsername(username);
+
+    if(user && await bcrypt.compare(password, user.password)) {
+      const { password, username, ...rest } = user;
+      return rest;
+    }
+
+    return null;
+  }
+
+  async validateEmail(email: string, password: string): Promise<any> {
+    const user = await this.usersService.findOneEmail(email);
+
+    if(user && await bcrypt.compare(password, user.password)) {
+      const { password, email, ...rest } = user;
+      return rest;
+    }
+
+    return null;
+  }
+
+  loginUsername(project_id: string, username: string, password: string) {
+    const payload = { username: username, sub: project_id };
+
+    return { access_token: this.jwtService.sign(payload, { expiresIn: process.env.JWT_EXPIRATION }) };
```

Comment:
NestJS has a good module for configurations that may be worth considering. The benefit of using their configuration tool is that is makes it easier to swap out configurations, makes it easier to have defaults, and can explicitly throw errors if a config is not found.

https://docs.nestjs.com/techniques/configuration